### PR TITLE
[UXE-156] feat: password text field component

### PR DESCRIPTION
**Pull request by jakeaustinlewis**
[UXE-156](https://resolvn.atlassian.net/browse/UXE-156)

### Summary
This PR introduces the `PasswordTextField` component, an enhanced variant of the Material UI `TextField`. It is designed to add privacy and enhance the user experience of typing sensitive information in a text field. This component incorporates a toggleable visibility button that allows users to hide or reveal sensitive information when clicked.

image

[UXE-156]: https://resolvn.atlassian.net/browse/UXE-156?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/src/PasswordTextField/PasswordTextField.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    if (isVisible) {
+      return 'text'
+    }
+
+    return 'password'
+  }, [isVisible])
+
+  return (
+    <PasswordTextFieldRoot
+      ownerState={props} // do I need this?
```
Comment (0x706b) [inline packages/components/src/PasswordTextField/PasswordTextField.tsx:140]: Yes, this is so that the theme can get the props 🙂

Comment (0x706b) [reply packages/components/src/PasswordTextField/PasswordTextField.tsx:140]: Can you delete the comment?

File: packages/components/src/PasswordTextField/PasswordTextField.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    toggleVisibility,
+  ])
+
+  const getType = React.useMemo(() => {
+    if (isVisible) {
+      return 'text'
+    }
+
+    return 'password'
+  }, [isVisible])
```
Comment (0x706b) [inline packages/components/src/PasswordTextField/PasswordTextField.tsx:136]: `getType` sounds like it should be a function, but it's a string here. Probably just `type` would be fine

File: packages/components/src/PasswordTextField/PasswordTextField.tsx (hunk 1 above)
Comment (jakeaustinlewis) [reply packages/components/src/PasswordTextField/PasswordTextField.tsx:140]: Thank you 🙂

Comment (jakeaustinlewis) [reply packages/components/src/PasswordTextField/PasswordTextField.tsx]: Deleted 👍

File: packages/components/src/PasswordTextField/PasswordTextField.tsx (hunk 2 above)
Comment (jakeaustinlewis) [reply packages/components/src/PasswordTextField/PasswordTextField.tsx]: Okay, sounds good

**Review by steve-jones — APPROVED** [review]

**Pull request merged**